src/hooks/UrqlUseMutation.re

```diff
@@ -61,3 +61,21 @@ let useMutation = (~request) => {
 
   (response, executeMutation);
 };
+
+let useDynamicMutation = (~query, ~parse) => {
+  let (responseJs, executeMutationJs) = useMutationJs(query);
+
+  let response =
+    React.useMemo1(
+      () => responseJs |> urqlResponseToReason(parse),
+      [|responseJs|],
+    );
+
+  let executeMutation =
+    React.useCallback1(
+      varsObj => executeMutationJs(Some(varsObj##variables)),
```

Comment:
A nit on naming here. This is the `Js.t` created by calling `.make()` on the `graphql_ppx_re` module, yeah? Could we use the same nomenclature we do in `useMutation` and name this `request`?

```suggestion
      request => executeMutationJs(Some(request##variables)),
```

Reply:
The type is slightly different (explained below), which is why I went with a different name. Doesn't bother me to change it though, if you think it's clearer.

Reply:
Ah that makes sense. I really wish we could destructure "open" objects like this so it'd be more like:

```reason
{ variables } => executeMutationJs(Some(variables))
```

Still I do find `request` a little easier to understand. This object passed to this callback is the result of calling `.make()` on that `graphql_ppx_re` module, and we have been calling that `request` throughout the codebase. Even tho we're restricting its type a bit more, it's more or less the same thing.